crates/graphql-composition/src/compose/scalar.rs

```diff
@@ -5,6 +5,9 @@ pub(crate) fn merge_scalar_definitions<'a>(
     definitions: &[DefinitionWalker<'a>],
     ctx: &mut Context<'a>,
 ) {
+    if first.name().as_str() == "join__FieldSet" {
```

Comment:
I wonder if we should have special treatment in a general way for scalars defined by federation directives (there are others IIRC)

Reply:
yeah think so, this was really adhoc